modules/extensions/src/main/java/org/apache/synapse/mediators/bsf/ScriptMediator.java

```diff
@@ -335,7 +353,12 @@ private Object mediateWithExternalScript(MessageContext synCtx)
      * @throws ScriptException For any errors , when compile , run the script
      */
     private Object mediateForInlineScript(MessageContext synCtx) throws ScriptException {
-        ScriptMessageContext scriptMC = new ScriptMessageContext(synCtx, xmlHelper);
+        ScriptMessageContext scriptMC;
+        if (language.equals(NASHORN_JAVA_SCRIPT)) {
+            scriptMC = new NashornJavaScriptMessageContext(synCtx, xmlHelper);
+        } else {
+            scriptMC = new CommonScriptMessageContext(synCtx, xmlHelper);
+        }
```

Comment:
Look like we have duplicate of this code chunk in line 327. Better to extract a private method and use.